src/scripts/main.js

```diff
@@ -1,3 +1,129 @@
 'use strict';
 
-// write your code here
+function keyListener(e) {
+  if (game.getStatus() !== 'playing') {
+    return;
+  }
+
+  switch (e.key) {
+    case 'ArrowDown':
+      game.moveDown();
+      break;
+    case 'ArrowLeft':
+      game.moveLeft();
+      break;
+    case 'ArrowRight':
+      game.moveRight();
+      break;
+    case 'ArrowUp':
+      game.moveUp();
+      break;
+  }
+
+  update();
```

Comment:
The 'update' function is being called directly within the 'keyListener' function. Consider using an event-driven approach to trigger updates, which can improve the separation of concerns and make the codebase more maintainable.